monitor/cert_submitter.go

```diff
@@ -115,14 +117,24 @@ type certSubmitter struct {
 func (c *certSubmitter) run() {
 	go func() {
 		for {
-			c.submitCertificates()
-			c.logger.Printf("Sleeping for %s before next certificate submission\n",
-				c.certSubmitInterval)
-			c.clk.Sleep(c.certSubmitInterval)
+			select {
+			case <-c.stopChannel:
+				return
+			default:
```

Comment:
To get the semantics I think you want here, you should do:

```
select {
  case <-c.stopChannel:
    return
  case <-time.After(c.certSubmitInterval):
}
c.submitCertificates()
```

That will allow the stop channel to interrupt the sleep.

Reply:
Aha! That's better, thank you. Fixed in b8adee8